src/aspects/ec2-automated-shutdown/index.ts

```diff
@@ -0,0 +1,149 @@
+            return;
+        }
+        const stack = Stack.of(node);
+        this.createShutdownMechanism(node, stack);
+    }
+
+    private getLambdaFunction(stack: Stack): SecureFunction {
+        const stackId = stack.stackId;
+        if (!Ec2AutomatedShutdown.lambdaByStack.has(stackId)) {
+            const lambda = new SecureFunction(stack, 'Ec2ShutdownFunction', {
+                code: Code.fromAsset(join(__dirname, 'handler')),
+                runtime: Runtime.NODEJS_20_X,
+                handler: 'index.handler',
+                timeout: Duration.seconds(15),
+                encryption: this.props.encryption,
+                ...this.props.LambdaConfiguration
+            });
+
+            if (!Ec2AutomatedShutdown.policyByStack.has(stackId)) {
+                const policy = new PolicyStatement({
+                    actions: ['ec2:StopInstances'],
+                    effect: Effect.ALLOW
+                });
+                Ec2AutomatedShutdown.policyByStack.set(stackId, policy);
+                lambda.role.addToPolicy(policy);
+            }
+            Ec2AutomatedShutdown.lambdaByStack.set(stackId, lambda);
+        }
+        return Ec2AutomatedShutdown.lambdaByStack.get(stackId)!;
+    }
+
+    private addInstanceToPolicy(
+        instance: Instance | CfnInstance,
+        stack: Stack
+    ): void {
+        const stackId = stack.stackId;
+        const policy = Ec2AutomatedShutdown.policyByStack.get(stackId);
+
+        if (!policy) {
+            throw new Error('Policy not initialized');
+        }
+
+        const instanceArn = `arn:${Aws.PARTITION}:ec2:${Aws.REGION}:${Aws.ACCOUNT_ID}:instance/${
+            instance instanceof Instance ? instance.instanceId : instance.ref
+        }`;
+
+        policy.addResources(instanceArn);
+    }
+
+    private createShutdownMechanism(instance: IConstruct, stack: Stack): void {
+        const lambdaFunction = this.getLambdaFunction(stack);
+
+        let instanceId: string;
+        if (instance instanceof Instance) {
+            instanceId = instance.instanceId;
+        } else if (instance instanceof CfnInstance) {
+            instanceId = instance.ref;
+        } else {
+            throw new Error('Unsupported instance type');
+        }
+
+        this.addInstanceToPolicy(instance, stack);
+
+        const MetricConfig =
+            this.props.MetricConfig || this.defaultMetricConfig;
+
+        const metric = new Metric({
+            namespace: 'AWS/EC2',
+            metricName: MetricConfig.MetricName,
+            dimensionsMap: { InstanceId: instanceId },
+            period: MetricConfig.period,
+            statistic: MetricConfig.statistic
+        });
+
+        const alarm = new Alarm(
+            instance,
+            `${metric.metricName}-${instance.node.id}-${stack.node.addr}'`,
+            {
+                metric: metric,
+                threshold: MetricConfig.threshold,
+                evaluationPeriods: 2,
```

Comment:
Do you think this evaluationPeriods and comparisonOperator should be options given to the user? I wonder if we should rename the CustomMetricConfig to like.. AlarmConfig or ShutdownConfig and have it include these, ex:

```
export interface AlarmConfig {
    readonly metricName: string;
    readonly period: Duration;
    readonly statistic: string;
    readonly threshold: number;
    readonly evaluationPeriods?: number;
    readonly datapointsToAlarm?: number;
    readonly comparisonOperator?: ComparisonOperator;
}
```